pkg/crc/config/viper_config.go

```diff
@@ -39,34 +39,59 @@ func NewViperStorage(configFile, envPrefix string) (*ViperStorage, error) {
 	}, nil
 }
 
-func (c *ViperStorage) Get(key string) interface{} {
-	return c.viper.Get(key)
+func (c *ViperStorage) Get(key string) (interface{}, error) {
+	if err := c.refresh(); err != nil {
+		return nil, err
+	}
+	return c.viper.Get(key), nil
 }
 
 func (c *ViperStorage) Set(key string, value interface{}) error {
-	c.viper.Set(key, value)
-	return c.viper.WriteConfigAs(c.configFile)
+	in, err := ioutil.ReadFile(c.configFile)
```

Comment:
Not clear what problems this is solving, maybe `Set` was misused, but if this has been causing issues, this is not obvious at all what issues.

Reply:
`Set` is not made for writing a new configuration but for overriding. You can reload 10 times the configuration, the override will stay. Viper is really made for reading configuration, not writing. Here I keep viper only for Get.

Reply:
Could probably be expanded a bit on in the commit log, and does not hurt to quote viper documentation
```

// Set sets the value for the key in the override register.
// Set is case-insensitive for a key.
// Will be used instead of values obtained via
// flags, config file, ENV, default, or key/value store.
func Set(key string, value interface{}) { v.Set(key, value) }
```